cmake/modules/FindTensorFlow.cmake

```diff
@@ -196,9 +206,41 @@ if(${TF_DISABLE_ASSERTS})
   add_definitions("-DNDEBUG=True")
 endif()
 
+if(IS_DIRECTORY $ENV{TENSORFLOW_BUILD_DIR})
+  set(TensorFlow_BUILD_DIR "$ENV{TENSORFLOW_BUILD_DIR}" CACHE PATH "directory containing the file 'libtensorflow_cc.so'")
+else()
+  if(NOT IS_DIRECTORY ${TensorFlow_BUILD_DIR})
+    set(TensorFlow_BUILD_DIR "TensorFlow_BUILD_DIR-NOTFOUND" CACHE PATH "directory containing the file 'libtensorflow_cc.so'")
+  endif()
+endif()
+if(IS_DIRECTORY ${TensorFlow_BUILD_DIR})
+  message(STATUS "TensorFlow_BUILD_DIR is ${TensorFlow_BUILD_DIR}")
+else()
+  message(FATAL_ERROR "No TensorFlow_BUILD_DIR detected,\n"
+    "please specify the path in ENV 'export TENSORFLOW_BUILD_DIR=...'\n or cmake -DTensorFlow_BUILD_DIR:PATH=...\n"
+    "to the directory containing the file 'libtensorflow_cc.so'"
+    )
+endif()
+
+if(IS_DIRECTORY $ENV{TENSORFLOW_SOURCE_DIR})
+  set(TensorFlow_SOURCE_DIR "$ENV{TENSORFLOW_SOURCE_DIR}" CACHE PATH "source code for tensorflow (i.e. the git checkout directory of the source code)")
+else()
+  if(NOT IS_DIRECTORY ${TensorFlow_SOURCE_DIR})
+    set(TensorFlow_SOURCE_DIR "TensorFlow_SOURCE_DIR-NOTFOUND" CACHE PATH "source code for tensorflow (i.e. the git checkout directory of the source code)")
+  endif()
+endif()
+if(IS_DIRECTORY ${TensorFlow_SOURCE_DIR})
```

Comment:
Shouldn't this be part of the else branch of `if(IS_DIRECTORY $ENV{TENSORFLOW_SOURCE_DIR})` above?